TCAT/Controllers/HomeOptionsCardViewController.swift

```diff
@@ -191,33 +205,37 @@ class HomeOptionsCardViewController: UIViewController {
         }
     }
 
-    func createDefaultSections() {
-        var allSections: [Section] = []
-        let recentSearchesSection = Section.recentSearches(items: recentLocations)
-        var favoritesSection = Section.favorites(items: favorites)
-        if favoritesSection.isEmpty {
-            let addFavorites = Place(
-                name: Constants.General.firstFavorite,
-                placeDescription: Constants.General.tapHere)
-            addFavorites.type = .busStop // Special exception to make pin blue for favorite!
-            favoritesSection = Section.favorites(items: [addFavorites])
+    /// Updates the table sections to show favorites, recent searches, and all stops. Ultimately reloads the table.
+    func updateSections() {
+        sections.removeAll()
+        sections.append(Section.favorites(items: favorites))
+        if recentLocations.count > 0 {
+            sections.append(Section.recentSearches(items: recentLocations))
         }
-        allSections.append(favoritesSection)
-        allSections.append(recentSearchesSection)
-        allSections.append(Section.seeAllStops)
-        sections = allSections.filter { !$0.isEmpty || $0 == Section.seeAllStops }
+        sections.append(Section.seeAllStops)
+    }
+    
+    /// Updates recent searches and favorites. Ultimately reloads the table.
+    func updatePlaces() {
+        recentLocations = Global.shared.retrievePlaces(for: Constants.UserDefaults.recentSearch)
+        favorites = Global.shared.retrievePlaces(for: Constants.UserDefaults.favorites)
     }
 
+    // MARK: - Table Calculations
+    
     private func tableViewContentHeight() -> CGFloat {
         return sections.reduce(0) { (result, section) -> CGFloat in
-            var sectionHeaderHeight: CGFloat = 0
             switch section {
-            case .favorites, .recentSearches: sectionHeaderHeight = headerHeight
-            case .seeAllStops: sectionHeaderHeight = HeaderView.separatorViewHeight
-            default: break
+            case .favorites:
+                let rows = CGFloat(max(1, section.getItems().count)) // at least the Add Favorites row if there are none
```

Comment:
Lets change comment to `// We should always have the Add Favorites row` and put it before line 230